src/combinators/mod.rs

```diff
@@ -1017,6 +1017,24 @@ impl BackToTheBuffer for &mut [u8] {
     }
 }
 
+impl BackToTheBuffer for Vec<u8> {
+    fn reserve_write_use<Tmp, Gen: Fn(Self) -> GenResult<(Self, Tmp)>, Before: Fn(Self, Tmp) -> GenResult<Self>>(mut self, reserved: usize, gen: Gen, before: Before) -> GenResult<Self> {
+        let start_len = self.len();
+        self.reserve(reserved);
+        unsafe { self.set_len(start_len + reserved) };
+        let tmp_vec = unsafe { Vec::from_raw_parts(self[start_len..].as_mut_ptr(), 0, reserved) };
```

Comment:
This is going to crash if the `before()` function tries to write more than `reserved`. `Vec` would re-allocated and free the pointer in the middle of the original vector.